### DISCUSSION TITLE
Add @Inherited to @XrayTest annotation

### DISCUSSION THREAD

**Discussion by Michihalo** [Xray-App/xray-junit-extensions#52] [Ideas]
As the title suggests, it would be nice to add the @Inherited annotation to the XrayTest annotation, this would allow to create custom annotations to be executed in pipelines only when specifically needed.

e.g.:
```
@XrayTest
@Test
@EnabledOnOs(OS.LINUX)
@Target({ ElementType.METHOD })
@Retention(RetentionPolicy.RUNTIME)
public @interface LinuxXrayTest {
    @AliasFor (
        annotation = XrayTest.class,
        attribute = "key"
    )
    String key() default = "";
}
```
![image](https://github.com/Xray-App/xray-junit-extensions/assets/42893641/ec996cb7-8298-4e0b-bf1f-010c3d0f0717)
The AnnotationSupport.findAnnotation method used in the XmlReportWriter is not able to find the XrayTest annotation in this case.

**Comment by bitcoder** [comment]
Hi @Michihalo, is that `@AliasFor` from Spring?
Can you provide a test class or a small repo showcasing this with a concrete example?

**Reply by bitcoder** [reply]
@Michihalo  Can you provide further feedback?

**Reply by bitcoder** [reply]
any feedback @Michihalo ?
